vars/runATH.groovy

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env groovy
+
+/**
+ * Wrapper for running the ATH see README and runATH.txt for full documentation
+ */
+
+def call(Map params = [:]) {
+    def athUrl = params.get('athUrl', 'https://github.com/jenkinsci/acceptance-test-harness.git')
+    def athRevision = params.get('athRevision', 'master')
+    def metadataFile = params.get('metadataFile', 'essentials.yml')
+    def jenkins = params.get('jenkins', 'latest')
+
+    def mirror = "http://mirrors.jenkins.io/"
+    def defaultCategory = "org.jenkinsci.test.acceptance.junit.SmokeTest"
+    def jenkinsURl = jenkins
+    def metadata
+    def athContainerImage
+    def isLocalATH
+    def isVersionNumber
+
+    def athSourcesFolder = "athSources"
+
+    def supportedBrowsers = ["firefox"]
+
+    def skipExecution = false
+
+    def localPluginsStashName = env.RUN_ATH_LOCAL_PLUGINS_STASH_NAME ?: "localPlugins"
+
+    ensureInNode(env, env.RUN_ATH_SOURCES_AND_VALIDATION_NODE ?: "linux", {
+        List<String> env = [
+                "JAVA_HOME=${tool 'jdk8'}",
+                'PATH+JAVA=${JAVA_HOME}/bin',
+                "PATH+MAVEN=${tool 'mvn'}/bin"
+
+        ]
+
+        if (!fileExists(metadataFile)) {
+            echo "Skipping ATH execution because the metadata file does not exist. Current value is ${metadataFile}."
+            skipExecution = true
+            return
+        }
+
+        stage("Getting ATH sources and Jenkins war") {
+            // Start validation
+            metadata = readYaml(file: metadataFile).ath
```

Comment:
Didn't you mean using `readYaml(file: metadataFile)?.ath` instead? I think it won't even reach the null check next line if the file is absent, but maybe it's not supposed to happen indeed?

Reply:
Yeah, a blank file must fail, but better have a good error message than an NPE. If the file is absent the intention is to work as a no-op and is managed before this lines